From 6750cc564a17c812555cca587660240ccffaaed3 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 4 Apr 2015 08:50:40 -0700
Subject: [PATCH] threads: Define pthread* definitions for glibc compliant libs

This code is assuming glibc but not explicitly saying it
so lets make it so. Fixes following on musl

threads.c:80:27: error: macro "pthread_equal" requires 2 arguments, but
only 1 given
|  extern int pthread_equal ()

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending
---
 threads.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: libxml2-2.9.2/threads.c
===================================================================
--- libxml2-2.9.2.orig/threads.c
+++ libxml2-2.9.2/threads.c
@@ -47,7 +47,7 @@
 #ifdef HAVE_PTHREAD_H
 
 static int libxml_is_threaded = -1;
-#ifdef __GNUC__
+#if defined(__GNUC__) && defined(__GLIBC__)
 #ifdef linux
 #if (__GNUC__ == 3 && __GNUC_MINOR__ >= 3) || (__GNUC__ > 3)
 extern int pthread_once (pthread_once_t *__once_control,
